### ISSUE TITLE
Version of Java in Codespaces image no longer works with the Java extension

### ISSUE THREAD

**Issue by Chuxel** [microsoft/vscode-dev-containers#495]
The Java extension now requires JDK 11, while the OS version of Java is only 8.  It is possible to do JDK 8 work if both JDKs are installed and the exact right configuration is done in VS Code.  However, this is requires very specific devcontianer.json config. (See https://github.com/redhat-developer/vscode-java/issues/1543)

To resolve this problem, we either need to:

1. Install OpenJDK 11 and add it as the new JAVA_HOME
2. Or drop Java support in the main Codespaces Linux image in favor of the pre-built Java one being PR'd in #493.

//cc: @edgonmsft @lostintangent @jkeech @nikmd23 @matthewisabel

**Comment by akaroml** [comment]
@Chuxel the Java 11 is only required by the Java Language Server itself. Users can still work with Java 8 or any other Java version in their projects. So the use case of having both Java 8 and 11 installed should work for sure.

@testforstephen should be able to help.

**Comment by testforstephen** [comment]
This requirement only asks you to install JDK 11 to run the vscode-java extension itself, because it's also a Java program and its code is compiled to Java 11 version. If you want to run a Java 8 application in dev container, you need install two JDKs and configure the dual JDKs in the user setting below.
```
{
    "java.home": "C:\\AdoptOpenJDK\\jdk-11.0.8.10-hotspot", // This is used to launch vscode-java extension itself.
    "java.configuration.runtimes": [
        {
          "name": "JavaSE-1.8",
          "path": "C:\\Java\\openjdk8u181-b13" // This tells Java extension to use the specified JDK to run Java 8 application.
        }
    ],
}
```

**Comment by Chuxel** [comment]
Yeah we'll have to create a dedicated definition for it. We can onboard it into image build as well - but I queued up JDK 11/14 for that in the mean time.  The image build system supports multiple definitions contributing to the same repository for this reason.

**Comment by Chuxel** [comment]
I've got a PR out that will tactically add JDK 11 in and make it the default (and leave the existing OS installed JDK 8).  Oryx will be adding JDK 8/11 into that image as well so this would be a point in time thing.

**Comment by edgonmsft** [comment]
By the way another thing is that the git compile/installation we have is because the version of git included in the image is too old. @dmgardiner25 has more details I think.

**Comment by dmgardiner25** [comment]
We were using the latest version available to us on `stretch` but users were requesting features only available in `buster`. Once Oryx updates the image to use `buster` (which they haven't given a timeline for but have said that it's in the works), we'll no longer have to manually build git.

**Comment by Chuxel** [comment]
> We were using the latest version available to us on stretch but users were requesting features only available in buster. Once Oryx updates the image to use buster (which they haven't given a timeline for but have said that it's in the works), we'll no longer have to manually build git.

Good to know.  Can add it back with a note. Let's discuss this more on the PR. https://github.com/microsoft/vscode-dev-containers/pull/500

**Comment by Chuxel** [comment]
This particular problem is resolved with #500. I just raised #517 to get a stand-alone JDK 8 definition with the needed settings in it.
